lib/php/rendition-factory/src/Transformer/Video/VideoToAnimationTransformerModule.php

```diff
@@ -2,22 +2,104 @@
 {
+    public function __construct(#[AutowireLocator(FormatInterface::TAG, defaultIndexMethod: 'getFormat')] private ServiceLocator $formats,
+        private ModuleOptionsResolver $optionsResolver,
+    ) {
+    }
+
     public static function getName(): string
     {
         return 'video_to_animation';
     }
 
+    public static function getDocumentation(): Documentation
+    {
+        static $doc = null;
+        if (null === $doc) {
+            $treeBuilder = Documentation::createBaseTree(self::getName());
+            self::buildConfiguration($treeBuilder->getRootNode()->children());
+            $doc = new Documentation(
+                $treeBuilder,
+                <<<HEADER
+                Converts a video to an animated gif / png.
+                HEADER
+            );
+        }
+
+        return $doc;
+    }
+
+    private static function buildConfiguration(NodeBuilder $builder): void
+    {
+        // @formatter:off
+        $builder
+            ->arrayNode('options')
+                ->info('Options for the module')
+                ->children()
+                    ->scalarNode('start')
+                        ->defaultValue(0)
+                        ->info('Start time in seconds or timecode')
+                        ->example('2.5 ; "00:00:02.50" ; "{{ attr.start }}"')
+                    ->end()
+                    ->scalarNode('duration')
+                        ->defaultValue(null)
+                        ->info('Duration in seconds or timecode')
+                        ->example('30 ; "00:00:30.00" ; "{{ input.duration/2 }}"')
+                    ->end()
+                    ->scalarNode('fps')
+                        ->defaultValue(1)
+                        ->info('Frames per second')
+                    ->end()
+                    ->scalarNode('width')
+                        ->defaultValue(null)
+                        ->info('Width in pixels')
+                    ->end()
+                    ->scalarNode('height')
+                        ->defaultValue(null)
+                        ->info('Height in pixels')
+                    ->end()
+                    ->enumNode('mode')
+                        ->values([
+                            FFMpeg\Filters\Video\ResizeFilter::RESIZEMODE_INSET,
+                            // todo: implement other modes
+                            // FFMpeg\Filters\Video\ResizeFilter::RESIZEMODE_FIT,
+                            // FFMpeg\Filters\Video\ResizeFilter::RESIZEMODE_SCALE_WIDTH,
+                            // FFMpeg\Filters\Video\ResizeFilter::RESIZEMODE_SCALE_HEIGHT,
+                        ])
+                        ->defaultValue(FFMpeg\Filters\Video\ResizeFilter::RESIZEMODE_INSET)
+                        ->info('Resize mode')
+                    ->end()
+                    ->scalarNode('format')
+                        ->isRequired()
+                        ->info('Output format')
```

Comment:
Would be great to have available formats here

Reply:
yes. "Formats" are now top level (/src) and should be used in all modules (now only video).
Since its tagged, Il will add a automatic list on the doc generator.

Reply:
Not your actual FormatInterface which is linked to FFMPEG only, see your interface methods!